sim/dex-abilities.ts

```diff
@@ -76,6 +77,8 @@ export class Ability extends BasicEffect implements Readonly<BasicEffect> {
 	}
 }
 
+const EMPTY_ABILITY = Utils.deepFreeze(new Ability({name: '', exists: false}));
+
 export class DexAbilities {
 	readonly dex: ModdedDex;
 	readonly abilityCache = new Map<ID, Ability>();
```

Comment:
Would there be any point prefilling the empty ability in the ability cache? (Similarly for the other caches.)
```suggestion
	readonly abilityCache = new Map<ID, Ability>([['', EMPTY_ABILITY]]);
```

Reply:
My only concern there would be that abilityCache et. all used to only contain objects with `.exists=true`.
I don't know whether that invariant actually matters, though, since iterating over abilityCache doesn't seem too useful anyway given that aliases can be present.
Something to consider.